tests/helpers/pattern.py

```diff
@@ -165,3 +167,9 @@ def __ne__(self, other):
 
 ANY.int = ANY.such_that(lambda x: isinstance(x, numbers.Integral))
 ANY.int.name = 'ANY.int'
+
+# Note: in practice it could be any int32, but as in those cases we expect the number to be
+# incremented sequentially, this should be reasonable for tests.
+ANY.small_int = ANY.such_that(lambda x: isinstance(x, numbers.Integral) and x < 10000)
```

Comment:
Since it's signed, this should check the negative bound as well, just in case.

Also, since it's for IDs specifically, maybe make it `ANY.dap_id`?